### ISSUE TITLE
regression in gulp-less dependency (less) causes gulp build task to fail

### ISSUE THREAD

**Issue by canavandl** [bokeh/bokeh#4299]
A new release from the "less" project (that is a dependency of gulp-less) contains a bug in sourcemap generation. This causes our build process to fail (see traceback below).

Proposed fix: commenting out LESS sourcemap generation until a new "less" release is made.

```
[11:25:18] Using gulpfile ~/dev/bokeh/bokehjs/gulpfile.js
[11:25:18] Starting 'typings:install'...
[11:25:19] Finished 'typings:install' after 740 ms
[11:25:19] Starting 'build'...
[11:25:19] Starting 'scripts'...
[11:25:19] Starting 'scripts:coffee'...
[11:25:19] Starting 'scripts:eco'...
[11:25:19] Starting 'scripts:ts'...
[11:25:19] Starting 'styles'...
[11:25:19] Starting 'styles:build'...
[11:25:20] Finished 'scripts:eco' after 953 ms
Potentially unhandled rejection [2] Error: Invalid mapping: {"generated":{"line":13,"column":2},"source":"/Users/lcanavan/dev/bokeh/bokehjs/src/vendor/bootstrap-3.1.1/less/grid.less","original":{"line":1,"column":null},"name":null}
  at SourceMapGenerator_validateMapping [as _validateMapping] (/Users/lcanavan/dev/bokeh/bokehjs/node_modules/gulp-less/node_modules/less/node_modules/source-map/lib/source-map-generator.js:277:13)
  at SourceMapGenerator_addMapping [as addMapping] (/Users/lcanavan/dev/bokeh/bokehjs/node_modules/gulp-less/node_modules/less/node_modules/source-map/lib/source-map-generator.js:101:12)
  at [object Object].SourceMapOutput.add (/Users/lcanavan/dev/bokeh/bokehjs/node_modules/gulp-less/node_modules/less/lib/less/source-map-output.js:88:42)
  at [object Object].Element.genCSS (/Users/lcanavan/dev/bokeh/bokehjs/node_modules/gulp-less/node_modules/less/lib/less/tree/element.js:42:12)
  at [object Object].Selector.genCSS (/Users/lcanavan/dev/bokeh/bokehjs/node_modules/gulp-less/node_modules/less/lib/less/tree/selector.js:104:21)
  at [object Object].Ruleset.genCSS (/Users/lcanavan/dev/bokeh/bokehjs/node_modules/gulp-less/node_modules/less/lib/less/tree/ruleset.js:389:21)
  at [object Object].Directive.outputRuleset (/Users/lcanavan/dev/bokeh/bokehjs/node_modules/gulp-less/node_modules/less/lib/less/tree/directive.js:124:18)
  at [object Object].Media.genCSS (/Users/lcanavan/dev/bokeh/bokehjs/node_modules/gulp-less/node_modules/less/lib/less/tree/media.js:34:10)
  at [object Object].Ruleset.genCSS (/Users/lcanavan/dev/bokeh/bokehjs/node_modules/gulp-less/node_modules/less/lib/less/tree/ruleset.js:414:18)
  at [object Object].SourceMapOutput.toCSS (/Users/lcanavan/dev/bokeh/bokehjs/node_modules/gulp-less/node_modules/less/lib/less/source-map-output.js:126:24)
  at [object Object].SourceMapBuilder.toCSS (/Users/lcanavan/dev/bokeh/bokehjs/node_modules/gulp-less/node_modules/less/lib/less/source-map-builder.js:23:35)
  at [object Object].ParseTree.toCSS (/Users/lcanavan/dev/bokeh/bokehjs/node_modules/gulp-less/node_modules/less/lib/less/parse-tree.js:33:47)
  at /Users/lcanavan/dev/bokeh/bokehjs/node_modules/gulp-less/node_modules/less/lib/less/render.js:31:40
  at /Users/lcanavan/dev/bokeh/bokehjs/node_modules/gulp-less/node_modules/less/lib/less/parse.js:63:17
  at Object.finish [as _finish] (/Users/lcanavan/dev/bokeh/bokehjs/node_modules/gulp-less/node_modules/less/lib/less/parser/parser.js:183:28)
  at Object.ImportVisitor._onSequencerEmpty (/Users/lcanavan/dev/bokeh/bokehjs/node_modules/gulp-less/node_modules/less/lib/less/visitors/import-visitor.js:35:14)
  at ImportSequencer.tryRun (/Users/lcanavan/dev/bokeh/bokehjs/node_modules/gulp-less/node_modules/less/lib/less/visitors/import-sequencer.js:50:14)
  at /Users/lcanavan/dev/bokeh/bokehjs/node_modules/gulp-less/node_modules/less/lib/less/visitors/import-sequencer.js:19:25
  at fileParsedFunc (/Users/lcanavan/dev/bokeh/bokehjs/node_modules/gulp-less/node_modules/less/lib/less/import-manager.js:50:17)
  at /Users/lcanavan/dev/bokeh/bokehjs/node_modules/gulp-less/node_modules/less/lib/less/import-manager.js:113:21
  at finish (/Users/lcanavan/dev/bokeh/bokehjs/node_modules/gulp-less/node_modules/less/lib/less/parser/parser.js:183:28)
  at Object.Parser.parse (/Users/lcanavan/dev/bokeh/bokehjs/node_modules/gulp-less/node_modules/less/lib/less/parser/parser.js:191:24)
  at loadFileCallback (/Users/lcanavan/dev/bokeh/bokehjs/node_modules/gulp-less/node_modules/less/lib/less/import-manager.js:112:64)

[11:25:21] Finished 'scripts:ts' after 2.33 s
[11:25:24] Finished 'scripts:coffee' after 4.83 s
[11:25:24] Starting 'scripts:compile'...
[11:25:24] Finished 'scripts:compile' after 7.62 μs
[11:25:24] Starting 'scripts:build'...
[11:25:28] Finished 'scripts:build' after 4.01 s
[11:25:28] Starting 'scripts:minify'...
```

**Comment by birdsarah** [comment]
The following diff pins less and works to prevent the error introduced by the regression in less. WE could add a note on the same line to remove the pin when the regression's fixed.

```diff
diff --git a/bokehjs/package.json b/bokehjs/package.json
index 472d56a..746289a 100644
--- a/bokehjs/package.json
+++ b/bokehjs/package.json
@@ -41,6 +41,7 @@
     "gulp-newer": "^1.1.0",
     "gulp-if": "^2.0.0",
     "merge": "^1.2.0",
+    "less": "2.6.1",
     "logtrap": "^0.0.1",
     "hammerjs": "^2.0.4",
     "mocha": "^2.2.5",
```

This seems a more intuitive fix than removing some of our build needs.

**Comment by canavandl** [comment]
a patched release (less=2.7.1) has been published to npm and resolves this issue.
